src/librustc/lint/mod.rs

```diff
@@ -92,6 +92,10 @@ macro_rules! lint_initializer {
 /// Declare a static item of type `&'static Lint`.
 #[macro_export]
 macro_rules! declare_lint {
+    (pub(crate) $name:ident, $level:ident, $desc:expr) => (
+        pub(crate) static $name: &'static ::rustc::lint::Lint
+            = &lint_initializer!($name, $level, $desc);
+    );
```

Comment:
Ugh, it requires `#![feature(macro_vis_matcher)]` in all crates where the macro containing it is used, including tests.